### DISCUSSION TITLE
Вопрос по API, которое используется в расширении

### DISCUSSION THREAD

**Discussion by kisoid** [Chimildic/YaMuTools#82] [Вопрос/ответ] [answered]
Когда обращаешься к истории прослушанного, выдаёт ли API таймстэмпы прослушиваний? Или просто список треков? То есть можно ли узнать дату/время конкретного прослушивания?

Спрашиваю, потому что интересно, будет ли возможность полноценно сохранять историю, если вдруг LastFM окончательно отвалится... Сегодня у меня их главная страница перестала открываться. Роскомнадзор...

**Comment by Chimildic** [comment, marked as answer]
API сайта не дает даты прослушиваний. Её можно получить через [gooex](https://chimildic.github.io/gooex/) функцией `Context.getRecentTracks()`. Там другое API и способ авторизации.

Также можно создать бэкап истории last.fm. Есть сервисы, которые дают выгрузить `.csv` или `.json` файлы. Бэкап по расписанию можно создать через библиотеку для GAS - [lastfm-backup](https://github.com/Chimildic/lastfm-backup).

**Reply by kisoid** [reply]
Ясно. Спасибо за инфу.

Ну, бэкап LastFM я делаю уже давно каждую 1000 прослушиваний... Попасть на сайт - не проблема. Просто у них зачастили 500 ошибки последнее время, а это тревожный сигнал (симптом финансовых проблем).

Плейлист "Мне нравится" на ЯМузыке тоже бэкаплю каждый вечер своим скриптом из планировщика. Опять же, чтоб быть в курсе, какие треки исчезают.

Надеюсь, в Яндексе вменяемые люди и хотя бы функцию скробблинга будут поддерживать. Потому что, если отключат, то вот это будет крайне печально.

**Reply by Chimildic** [reply]
Есть сторонние приложения для ПК и смартфонов для скробблинга на ласт.фм. Большая проблема трафик донести с таких устройств.

**Reply by kisoid** [reply]
Скробблинг, насколько мне известно, делается через обращения к субдомену *.audioscrobbler.com
Разве его тоже блокируют? Вроде только last.fm
